src/theory/uf/equality_engine_types.h

```diff
@@ -63,20 +63,24 @@ static const EqualityEdgeId null_edge = (EqualityEdgeId)(-1);
  * or a merge of an equality to false due to both sides being
  * (different) constants.
  */
-enum MergeReasonType {
-  /** Terms were merged due to application of congruence closure */
+enum MergeReasonType
+{
+  /** Terms were merged due to congruence */
   MERGED_THROUGH_CONGRUENCE,
-  /** Terms were merged due to application of pure equality */
+  /** Terms were merged due to an assumption */
   MERGED_THROUGH_EQUALITY,
-  /** Equality was merged to true, due to both sides of equality being in the same class */
+  /** Terms were merged due to reflexivity */
   MERGED_THROUGH_REFLEXIVITY,
-  /** Equality was merged to false, due to both sides of equality being a constant */
+  /** Terms were merged due to theory reasoning */
   MERGED_THROUGH_CONSTANTS,
-  /** (for proofs only) Equality was merged due to transitivity */
+  /** Terms were merged due to transitivity */
   MERGED_THROUGH_TRANS,
-
-  /** Reason types beyond this constant are theory specific reasons */
-  NUMBER_OF_MERGE_REASONS
+  /** Terms were merged due to arrays read-over-write */
```

Comment:
These shouldn't have to be added here.  Can you try to delete these as well?  Or perhaps at least mark "//!!!!!!!! temporary".  I believe I can delete them if you need.

Reply:
The array solver sends these reasons when it's asserting equalities. Previously it would dynamically generate these three, so there was some dedicated code for allowing for that. By making them static the handling of reasons is simpler.

Reply:
just write temporary then, I'll remove on proof-new.

Reply:
OK, done.